plugin-server/src/worker/ingestion/team-manager.ts

```diff
@@ -42,6 +44,29 @@ export class TeamManager {
         }
     }
 
+    async flushLastSeenAtCache(): Promise<void> {
+        const team_ids: string[] = []
+        const event_names: string[] = []
+        const last_seen_at_array: number[] = []
+
+        for (const event of this.eventLastSeenCache) {
+            team_ids.push(event[0].split('_')[0])
+            event_names.push(event[0].split('_')[1])
+            last_seen_at_array.push(event[1])
+        }
+
+        this.lastFlushAt = DateTime.now()
+        this.eventLastSeenCache = new Map()
+
+        await this.db.postgresQuery(
+            `UPDATE posthog_eventdefinition t1 SET t1.last_seen_at = GREATEST(t1.last_seen_at, t2.last_seen_at)
+            FROM (UNNEST ($1) as team_id, UNNEST($2) as name, UNNEST($3) as last_seen_at) as t2
+            WHERE t1.name = t2.name AND t1.team_id = t2.team_id`,
+            [team_ids, event_names, last_seen_at_array],
```

Comment:
This was brought up elsewhere, but why do we want track everything so precisely?

I don't see a business reason why you even need hourly precision here not to mention second-wise, and it makes everything perform worse. In this case we increase our memory usage and need to send more bytes over to postgres for the update.

It's also one of these decisions that are really hard to undo after the fast. I'd push really hard for using daily resolution

Reply:
Agreed! From a product perspective having daily resolution (`Date` instead of `DateTime` is enough). If we do move forward with this approach I'll update this to use only dates.